File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.cpp
// Last updated: 04/04/2026, 23:34:52
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        bool seen[1001] = {false};
        bool added[1001] = {false};
        for(int num : nums1) seen[num] = true;
        vector<int> ans;
        for(int num : nums2) {
            if(seen[num] && !added[num]) {
                ans.push_back(num);
                added[num] = true;
            }
        }
        return ans;
    }
};

File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.java
// Last updated: 04/04/2026, 23:34:56
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        int[] seen = new int[1001];
        for(int num : nums1) seen[num] = 1;
        List<Integer> ansList = new ArrayList<>();
        for(int num : nums2) {
            if(seen[num]==1) {
                ansList.add(num);
                seen[num]=2;
            }
        }
        int n = ansList.size();
        int[] ans = new int[n];
        for(int i=0;i<n;i++) ans[i] = ansList.get(i);
        return ans;
    }
}